### DISCUSSION TITLE
Is auto-upload-wifi-only useful?

### DISCUSSION THREAD

**Discussion by smichel17** [streetcomplete/StreetComplete#3478] [Ideas]
In https://github.com/streetcomplete/StreetComplete/discussions/3470#discussioncomment-1561623, @mnalis wrote (emphasis **mine**),

> Even as an advanced user, I sometimes forget, as unless SC was the foreground app when I got home, 
> **`auto-upload on wifi` is equivalent to `off`**.

The more I think about this, the more I agree with it. At least for myself:

- I am seldom on wifi when I am mapping. *Occasionally* I'm in a mall or something and *maybe* I'll connect to their wifi.
- Besides development, I rarely open SC at home. When I do, it's usually to download some data or plan a route before I go out.
- The first thing I did when I started mapping regularly was answer the quests near my house, so it's not like I usually have StreetComplete open near the start and ends of my walks, either.

The result is that I need to remember to open StreetComplete when I get home *anyway*. That already takes several taps; one more to press the upload button is not saving me any effort. And the setting doesn't make the upload button disappear (saving me screen space), either. So, I don't really see a purpose. I'd rather hit the button manually, and enjoy watching the score tick up!

I could imagine reworking this setting into something like "auto-upload: immediately / batched / never". Or just removing it (auto-upload becomes a toggle) and creating a new toggle, "remind me to upload quests" (only available when auto-upload is off).

Maybe other people use StreetComplete differently than me, but right now this option seems to accomplish ~nothing.

**Comment by mnalis** [comment]
Well, even for myself it depends. The point I was trying to make in that comment was that **_auto-wifi upload can't be counted on to do the job 100% of the time_** (or even near 100%).

Sure, I've mapped everything around my house too, but even so, auto-upload on wifi might work for me about half of the time. Ie. it depends if I was doing something with the phone between the last quest I mapped, and the time I got home. 

<details>

If I weren't doing anything else with the phone, SC would still be in foreground even if sitting in my pocket, so would (hopefully) upload the quests automatically when it got wifi. 
The problem would occur only when I had to do _something else on the phone_ after I cleared the area of SC quests, _and I knew there is nothing more to map_ - as I'd then likely remain in the other app I needed, instead of returning to SC.

</details>

Still, even if it looked more useless to me, I'm generally against removing features that were added before, even if their benefit looks somewhat low to me, unless the benefits reaped by removing the functionality is **huge** (like greatly simplifying codebase, or drastically reducing APK size) - which I don't think applies here.

The reasoning being that it is practically **guaranteed** there will be people who will feel **unhappy** when feature that was _working just fine for their use case_ gets removed for no apparent reason; and I prefer to spread happiness instead of sadness over the world.

<details>

Now, if the feature was never implemented before, and we were debating if it was worth the effort for creating such feature given known limitations of it, I'd more likely to be *"meh"*. As it is, if I were using that feature actively, I'd be more likely to suggest that SC implement that feature properly (eg. detecting wifi even if SC is not foreground app), however:
- I'm _usually_ not using that feature, because I prefer to decide manually when stuff happens (hey, on my computer, even plugged-in USB memory stick does not auto-mount)
- I know I would be opening SC soon enough, so forgetting to upload is not an issue for me
- I actually prefer SC (and vast majority of other android apps) gets completely terminated when I put some other app in foreground.
- doing stuff in the background seems (currently) to be fighting an uphill battle against Android manufacturers; and I think the benefit is not big enough to (likely continually) waste effort trying to fighting that. (especially since there are so many nice issues just waiting to be solved, like for example https://github.com/streetcomplete/StreetComplete/issues/3377 :smile: )

</details>

**Comment by westnordost** [comment]
Many people who do not have a flatrate still use internet on the phone very often because all over the city, there are lots of places with free wifi. Some friends of mine use the free wifi from the city's public transportation system all the time. So for them, I think this option very much makes sense.

On the other hand, of course, I'd expect them to usually have the "mobile data" option off pretty much always except when they really specifically need internet for something.

I'll ask him how he uses Streetcomplete.

**Reply by westnordost** [reply]
Hm actually he now got a flatrate (more or less) a year ago, so he just "has mobile data on and let's the app do its thing".

**Reply by mnalis** [reply]
Well, my internet plan is not flat, and I do keep mobile data off majority of the time (enabling it only to do specific action, like download SC area which I didn't manage to download before, or having to lookup some contact quickly on the web, and disabling it shortly afterward). 

Not incurring extra cost is one reason, but there are big other reasons too (battery life is quite big other one; not wanting to be traced where I am _all the time_ by Google and friends, or not wanting to be bothered by non-essential interruptions while I'm away from computer, are some of the other important reasons).

In majority of cases I'm quite content to wait until I get back on home wifi to use the Internet, so I rarely use public wifis (unless traveling abroad, when I both have much higher need for looking up information online, don't have a computer, and the mobile data costs are much higher).

**Reply by matkoniecz** [reply]
I used to have such plan some time ago but I also switched (I was happy about 300 MB/month but I switched due to limits on phone calls).

For people with strict data limits this would be not very useful as data downloads cannot be stopped, but at least allows to do photo uploads on wifi.

**Comment by Lee-Carre** [comment]
A scenario to consider would be something like a mobile-hotspot (connecting via 802.11, data relayed via mobile network). This is also relevant to vehicles; boats, planes, & trains often have 802.11 which relays via some other type of connection.

In my area, mobile data tariffs are ether not generous or not cheap. There are no flat-rate (unlimited use) tariffs, as far as I know. Even fixed-line connections are quite limited (typically 10GB per month, unless you're willing to pay significantly more).
But, then, being on an island means that all off-island traffic is via expensive submarine cables.

Many establishments offer 802.11 (especially in urban areas), usually without excess hurdles to connect. Including the hospital, library, a major bus route (#15), many hotels (and similar), most cafes (& similar).
Since they're on commercial tariffs (unlimited usage, but higher price) I oft make use of them since I need not be concerned about usage (and several offer much greater capacity than even 4G; the hospital is >100Mbps in each direction).
